src/coreclr/nativeaot/System.Private.CoreLib/src/System/ValueType.cs

```diff
@@ -95,21 +95,17 @@ public override unsafe bool Equals([NotNullWhen(true)] object? obj)
 
         public override unsafe int GetHashCode()
         {
-            int hashCode = (int)this.GetMethodTable()->HashCode;
+            HashCode hashCode = default;
+            hashCode.Add(this.GetMethodTable()->HashCode);
 
-            hashCode ^= GetHashCodeImpl();
-
-            return hashCode;
-        }
-
-        private unsafe int GetHashCodeImpl()
-        {
             int numFields = __GetFieldHelper(GetNumFields, out _);
 
             if (numFields == UseFastHelper)
-                return FastGetValueTypeHashCodeHelper(this.GetMethodTable(), ref this.GetRawData());
+                hashCode.Add(FastGetValueTypeHashCodeHelper(this.GetMethodTable(), ref this.GetRawData()));
```

Comment:
This should call hashCode.Add on the ReadOnlySpan created from the valuetype bytes (similar to what CoreCLR does).

Reply:
(And `FastGetValueTypeHashCodeHelper` should be deleted.)

Reply:
Ah ok, `FastGetValueTypeHashCodeHelper` is called from multiple places. I guess it can morph into a helper method that returns ReadOnlySpan<byte>